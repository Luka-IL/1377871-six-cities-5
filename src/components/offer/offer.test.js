import React from "react";
import renderer from "react-test-renderer";
import Offer from "./offer";

const AVATAR_URL = `https://api.adorable.io/avatars`;

const offers = [{
  pictures: [
    `http://farm66.static.flickr.com/65535/50440663037_e00f7f6e12_b.jpg`,
    `http://farm66.static.flickr.com/65535/50440663037_e00f7f6e12_b.jpg`,
    `http://farm66.static.flickr.com/65535/50440663037_e00f7f6e12_b.jpg`,
    `http://farm66.static.flickr.com/65535/50440663037_e00f7f6e12_b.jpg`,
    `http://farm66.static.flickr.com/65535/50440663037_e00f7f6e12_b.jpg`,
    `http://farm66.static.flickr.com/65535/50440663037_e00f7f6e12_b.jpg`
  ],
  title: `1Beautiful & luxurious studio at great location`,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua`,
  premium: false,
  type: `apartment`,
  rating: 5,
  bedrooms: 3,
  guests: 4,
  price: 20,
  households: [
    `WiFi`, `Heating`, `Kitchen`, `Cable TV`
  ],
  owner: {
    avatar: `${AVATAR_URL}/3}`,
    name: `Karl`,
    super: true
  },
  coordinates: [
    52.3909553944608, 4.85309666408176
  ],
  neighbourhood: [
    [52.3909553943508, 4.85309666406198],
    [52.3809553943508, 4.939309666406198],
    [52.369553943508, 4.85309666406198]
  ],
  comments: [
    {
      guest: {
        name: `Max`,
        avatar: `${AVATAR_URL}/2}`
      },
      description: `Consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua, ipsum dolor sit amet`,
      rating: 4,
      date: `April 2019`
    }, {
      guest: {
        name: `Lora`,
        avatar: `${AVATAR_URL}/3}`
      },
      description: `Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua, ipsum dolor sit amet`,
      rating: 3,
      date: `Mart 2018`
    }
  ]
}];

jest.mock(`../map/map`, () => `Map`);
jest.mock(`../list-neighbours/list-neighbours`, () => `ListNeighbours`);


it(`Should Offer render correctly`, () => {
  const tree = renderer
    .create(<Offer
      onMainClick={() => {}}
      onOfferClick={() => {}}
      offers={offers}
      offer={offers[0]}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
