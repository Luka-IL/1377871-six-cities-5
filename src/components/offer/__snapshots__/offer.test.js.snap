// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Offer render correctly 1`] = `
<div
  className="page"
>
  <Header />
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        />
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              1Beautiful & luxurious studio at great location
            </h1>
            <button
              className="property__bookmark-button button"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "80%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              apartment
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              20
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            />
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src=""
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
              </p>
              <p
                className="property__text"
              >
                An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.
              </p>
            </div>
          </div>
          <ListReviews
            id={1}
          />
        </div>
      </div>
      <section
        className="cities__map map"
        style={
          Object {
            "height": "20em",
            "margin": "auto",
            "width": "40em",
          }
        }
      >
        <Map />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighborhood
        </h2>
        <ListNeighbours
          onOfferClick={[Function]}
        />
      </section>
    </div>
  </main>
</div>
`;
